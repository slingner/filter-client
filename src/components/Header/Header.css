@import url('https://fonts.googleapis.com/css?family=Montserrat:400,600&display=swap');

@import url('https://fonts.googleapis.com/css?family=Delius&display=swap');

.header {
  display: flex;
  flex-direction: row;
}

.nav-bar > {
  color: black;
  display: flex;
  justify-content: left;
  margin-right: auto;
  list-style-type: none;
  text-decoration: none;
  font-family: 'Montserrat';
}

.nav-bar .Links {
  text-decoration: none;
  list-style-type: none;
  color: rgb(90, 90, 90);
  padding: 15px;
}

.Links:hover {
  cursor: pointer;
  color: rgb(227, 161, 85);
}

.nav-title {
  text-decoration: none;
  color: rgb(90, 90, 90);
  font-family: 'Delius';
  font-size: 40px;
}

.nav-title:hover {
  cursor: pointer;
  color: rgb(110, 147, 115);
  font-size: 45px;
}

@media screen and (max-width: 600px) {
  .Links {
    display: flex;
    flex-direction: column;
    background-color: rgb(197, 205, 187);
    margin: 5px;
    border-radius: 2px;
    padding: 1px;
  }

  .nav-bar {
    margin-top: 5px;
    min-width: 80%;
  }

  .nav-title {
    display: block;
    margin-top: 10px;
    font-size: 40px;
  }
}
