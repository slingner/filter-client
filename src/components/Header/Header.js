import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';
import TokenService from '../../services/token-service';
import { ReactComponent as FilterText } from './Assets/FILTER_TEXT.svg';
import { ReactComponent as FilterLogo } from '../../routes/HomePage/Assets/ICON.svg';

class Header extends Component {
  //if a user is logged in, this is what displays in NavBar
  renderLogoutLink = () => {
    return (
      <>
        <NavLink className="Links" to="/BeanList">
          All Beans
        </NavLink>
        <NavLink className="Links" to="/Account">
          Your Beans
        </NavLink>
        <NavLink className="Links" onClick={this.props.logoutClick} to="/">
          Logout
        </NavLink>
      </>
    );
  };
  //if a user is logged out, this is what displays in NavBar
  renderLoginLink = () => {
    return (
      <>
        <NavLink className="Links" to="/BeanList">
          All Beans
        </NavLink>
        <NavLink className="Links" to="/login">
          Log in
        </NavLink>
        <NavLink className="Links" to="/register">
          Register
        </NavLink>
      </>
    );
  };

  render() {
    return (
      <div className="headerWrapper">
        {/* <header className="header"> */}
        <NavLink to={'/'} className="nav-title">
          <FilterLogo className="logo" />
          <FilterText className="filterTextLogo" />
        </NavLink>
        {/* </header> */}
        <nav className="nav-bar">
          {/* hasAuthToken checks for log in or out state */}
          {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </nav>
      </div>
    );
  }
}

export default Header;
