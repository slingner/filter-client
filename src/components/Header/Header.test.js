import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe(`Header component`, () => {
  const props = {
  }

  it('renders the header by default', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the header', () => {
    const wrapper = shallow(<Header {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})