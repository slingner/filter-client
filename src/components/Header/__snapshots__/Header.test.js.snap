// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the header 1`] = `
<div
  className="headerWrapper"
>
  <NavLink
    className="nav-title"
    to="/"
  >
    <ForwardRef(SvgIcon)
      className="logo"
    />
    <ForwardRef(SvgFilterText)
      className="filterTextLogo"
    />
  </NavLink>
  <nav
    className="nav-bar"
  >
    <NavLink
      className="Links"
      to="/BeanList"
    >
      All Beans
    </NavLink>
    <NavLink
      className="Links"
      to="/login"
    >
      Log in
    </NavLink>
    <NavLink
      className="Links"
      to="/register"
    >
      Sign Up
    </NavLink>
  </nav>
</div>
`;

exports[`Header component renders the header by default 1`] = `
<div
  className="headerWrapper"
>
  <NavLink
    className="nav-title"
    to="/"
  >
    <ForwardRef(SvgIcon)
      className="logo"
    />
    <ForwardRef(SvgFilterText)
      className="filterTextLogo"
    />
  </NavLink>
  <nav
    className="nav-bar"
  >
    <NavLink
      className="Links"
      to="/BeanList"
    >
      All Beans
    </NavLink>
    <NavLink
      className="Links"
      to="/login"
    >
      Log in
    </NavLink>
    <NavLink
      className="Links"
      to="/register"
    >
      Sign Up
    </NavLink>
  </nav>
</div>
`;
